internal/ota-api/dto_test.go

```diff
@@ -0,0 +1,25 @@
+package otaapi
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestProgressBar_notCompletePct(t *testing.T) {
+	firmwareSize := int64(25665 * 2)
+	bar := formatStateData("fetch", "25665", &firmwareSize, false)
+	assert.Equal(t, "[=====     ] 50.00%", bar)
+}
+
+func TestProgressBar_ifFlashState_goTo100Pct(t *testing.T) {
+	firmwareSize := int64(25665 * 2)
+	bar := formatStateData("fetch", "25665", &firmwareSize, true) // If in flash status, go to 100%
+	assert.Equal(t, "[==========] 100.00%", bar)
```

Comment:
I haven't tried this new feature yet, but I find the progress bar a little bit shorter. What do you think of increasing it to 20 characters?

Reply:
Added a multiplier, so it can be resized as wish :)